### PR TITLE
Add test case for empty path

### PR DESCRIPTION
**Pull request by djfhe**
Add some test cases to ensure correct handling of an empty path.

### CODE REVIEW

File: test/index.test.ts [hunk 1]
```diff
@@ -110,6 +110,24 @@ it('returns undefined on non objects', () => {
   expect(getByPath(test, 'undef.baz')).toBe(undefined)
 })
 
+it('it returns undefined when accessing an unset path', () => {
+  const test = {
+    foo: 'bar',
+  } as const
+
+  // @ts-expect-error - '' is not define on the property, so it is correct that typescript throws an error here
```
Comment (saibotk) [inline test/index.test.ts:118]:
```suggestion
  // @ts-expect-error - '' is not defined on the object, so it is correct that typescript throws an error here
```

File: test/index.test.ts [hunk 2]
```diff
@@ -110,6 +110,24 @@ it('returns undefined on non objects', () => {
...
+  // but since no type information are available in runtime, getByPath will still try to retrieve its value
+  expect(getByPath(test, '')).toBe(undefined)
+
+  // @ts-expect-error - 'foo' is not set on the property, so typescript should throw an error here
+  // but as mentioned above, we still will set the value at runtime
+  setByPath(test, '', 'bar')
+
+  // @ts-expect-error - '' is not set on the property, so typescript should throw an error here
```
Comment (saibotk) [inline test/index.test.ts:126]:
```suggestion
  // @ts-expect-error - '' is not set on the object, so typescript should throw an error here
```

File: test/index.test.ts [hunk 3]
```diff
@@ -110,6 +110,24 @@ it('returns undefined on non objects', () => {
...
+    foo: 'bar',
+  } as const
+
+  // @ts-expect-error - '' is not define on the property, so it is correct that typescript throws an error here
+  // but since no type information are available in runtime, getByPath will still try to retrieve its value
+  expect(getByPath(test, '')).toBe(undefined)
+
+  // @ts-expect-error - 'foo' is not set on the property, so typescript should throw an error here
+  // but as mentioned above, we still will set the value at runtime
+  setByPath(test, '', 'bar')
```
Comment (saibotk) [inline test/index.test.ts:124]: this and the one line afterwards should rather get their own test case. The test name only implies that the access to an unset path is tested

File: test/index.test.ts [hunk 4]
```diff
@@ -110,6 +110,24 @@ it('returns undefined on non objects', () => {
   expect(getByPath(test, 'undef.baz')).toBe(undefined)
 })
 
+it('it returns undefined when accessing an unset path', () => {
```
Comment (saibotk) [inline test/index.test.ts:113]:
```suggestion
it('returns undefined when accessing an unset path', () => {
```

**Review by saibotk — CHANGES_REQUESTED** [review]

**Comment by djfhe** [comment]
i would close this PR in favor of https://github.com/clickbar/dot-diver/pull/11. I don't see any use in merging this PR right before 2.0 where we change this behavior anyways. @saibotk
